sdk/helper/pluginutil/run_config.go

```diff
@@ -0,0 +1,161 @@
+	cmd := exec.Command(rc.command, rc.args...)
+	cmd.Env = append(cmd.Env, rc.env...)
+
+	// Add the mlock setting to the ENV of the plugin
+	if rc.wrapper != nil && rc.wrapper.MlockEnabled() {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginMlockEnabled, "true"))
+	}
+	cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginVaultVersionEnv, version.GetVersion().Version))
+
+	if rc.isMetadataMode {
+		rc.logger = rc.logger.With("metadata", "true")
+	}
+	metadataEnv := fmt.Sprintf("%s=%t", PluginMetadataModeEnv, rc.isMetadataMode)
+	cmd.Env = append(cmd.Env, metadataEnv)
+
+	var clientTLSConfig *tls.Config
+	if !rc.autoMTLS && !rc.isMetadataMode {
+		// Get a CA TLS Certificate
+		certBytes, key, err := generateCert()
+		if err != nil {
+			return nil, err
+		}
+
+		// Use CA to sign a client cert and return a configured TLS config
+		clientTLSConfig, err = createClientTLSConfig(certBytes, key)
+		if err != nil {
+			return nil, err
+		}
+
+		// Use CA to sign a server cert and wrap the values in a response wrapped
+		// token.
+		wrapToken, err := wrapServerConfig(ctx, rc.wrapper, certBytes, key)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add the response wrap token to the ENV of the plugin
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginUnwrapTokenEnv, wrapToken))
+	}
+
+	secureConfig := &plugin.SecureConfig{
+		Checksum: rc.sha256,
+		Hash:     sha256.New(),
+	}
+
+	clientConfig := &plugin.ClientConfig{
+		HandshakeConfig:  rc.hs,
+		VersionedPlugins: rc.pluginSets,
+		Cmd:              cmd,
+		SecureConfig:     secureConfig,
+		TLSConfig:        clientTLSConfig,
+		Logger:           rc.logger,
+		AllowedProtocols: []plugin.Protocol{
+			plugin.ProtocolNetRPC,
+			plugin.ProtocolGRPC,
+		},
+		AutoMTLS: rc.autoMTLS,
+	}
+	return clientConfig, nil
+}
+
+func (rc runConfig) run(ctx context.Context) (*plugin.Client, error) {
+	clientConfig, err := rc.makeConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	client := plugin.NewClient(clientConfig)
+	return client, nil
+}
+
+type RunOpt func(*runConfig)
+
+func Env(env ...string) RunOpt {
+	return func(rc *runConfig) {
+		rc.env = append(rc.env, env...)
+	}
+}
+
+func Runner(wrapper RunnerUtil) RunOpt {
+	return func(rc *runConfig) {
+		rc.wrapper = wrapper
+	}
+}
+
+func PluginSets(pluginSets map[int]plugin.PluginSet) RunOpt {
+	return func(rc *runConfig) {
+		rc.pluginSets = pluginSets
+	}
+}
+
+func HandshakeConfig(hs plugin.HandshakeConfig) RunOpt {
+	return func(rc *runConfig) {
+		rc.hs = hs
+	}
+}
+
+func Logger(logger log.Logger) RunOpt {
+	return func(rc *runConfig) {
+		rc.logger = logger
+	}
+}
+
+func MetadataMode(isMetadataMode bool) RunOpt {
+	return func(rc *runConfig) {
+		rc.isMetadataMode = isMetadataMode
+	}
+}
+
+func AutoMTLS(autoMTLS bool) RunOpt {
+	return func(rc *runConfig) {
+		rc.autoMTLS = autoMTLS
+	}
+}
+
+func (r *PluginRunner) RunConfig(ctx context.Context, opts ...RunOpt) (*plugin.Client, error) {
```

Comment:
Nit: This might have been derived from the `runCommon` terminology and to keep things as close as it was before, but it's really creating a plugin client so something like `NewPluginClient` would be more appropriate.